### ISSUE TITLE
remove fork and exec modutils code, require kmod for module loading

### ISSUE THREAD

**Issue by cleech** [open-iscsi/open-iscsi#113]
Can we drop the code for running modprobe, and require kmod if you want module loading support?

**Comment by gonzoleeman** [comment]
I must admit we are not using libkmod for SUSE, but I'll work on changing that.

**Comment by gonzoleeman** [comment]
Note: your libkmod.h seems to be in <libkmod.h>, and our is in <kmod/libkmod.h>. So either an option has to be added to the Makefile so the compiler looks there, or the include will have to change depending on yet-another-define.

**Comment by cleech** [comment]
That sounds like pkgconfig checks, and possibly a real build system to handle distro differences

**Comment by gonzoleeman** [comment]
Agreed that both of those things would help smooth out the differences between our build systems, but I don't think this one thing is the final "straw", so to speak. Right now, it seems like we hide the differences between our build systems by distro-specific changes to (1) the SPEC file (i.e. how packages are built), and (2) the Makefiles. I would like to see these go away, because (among other thing) these act as a barrier to entry for non-main-distro use.

But in this case, I was able to handle this difference by a change to the %build rule in our SPEC file, in addition to a small change to usr/Makefile to put LDFLAGS at the end of the compile line.

I've had it on my list of things to do for a while to try to eliminate differences between our make files and the distro versions, but it's not very high on the list so doesn't get addressed.

Bottom line, I support a patch to remove the code supporting fork/exec now. :+1:

**Comment by gonzoleeman** [comment]
I've updated the newer SUSE versions of open-iscsi (and our github open-iscsi repo) to support libkmod now, so please go ahead with removal of this code.

**Comment by gonzoleeman** [comment]
Closed by pull request (merged) #138
